plugins/glustershd/rest.go

```diff
@@ -201,3 +201,95 @@ func selfHealHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func checkBrickExistence(volinfo *volume.Volinfo, hostname string, brickname string) bool {
+	bricks := volinfo.GetBricks()
+	for index := range bricks {
+		if bricks[index].Hostname == hostname && bricks[index].Path == brickname {
+			return true
+		}
+	}
+	return false
+}
+
+func splitBrainOperationHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	volname := mux.Vars(r)["volname"]
+	operation := mux.Vars(r)["operation"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req glustershdapi.SplitBrainReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrJSONParsingFailed)
+		return
+	}
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	// Validate volume type
+	if !isVolReplicate(volinfo.Type) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid operation for this volume type")
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrVolNotStarted)
+		return
+	}
+
+	var options []string
+	glusterdSockpath := path.Join(config.GetString("rundir"), "glusterd2.socket")
+	switch operation {
+	case "latest-mtime":
+		fallthrough
+	case "bigger-file":
+		if req.FileName != "" {
+			if !strings.HasPrefix(req.FileName, "/") {
+				restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "filename should be an absolute path in volume, should start with / notation")
+				return
+			}
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "please provide filename")
+			return
+		}
+		options = append(options, operation, req.FileName, "glusterd-sock", glusterdSockpath)
+
+	case "source-brick":
+		if req.HostName == "" || req.BrickName == "" {
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "please provide hostname and brickname")
+			return
+		}
+		if !checkBrickExistence(volinfo, req.HostName, req.BrickName) {
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "brick doesn't exist")
+			return
+		}
+		brickPath := fmt.Sprintf("%s:%s", req.HostName, req.BrickName)
+		if req.FileName != "" {
+			if !strings.HasPrefix(req.FileName, "/") {
+				restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "filename should be an absolute path in volume, should start with / notation")
+				return
+			}
+			options = append(options, operation, brickPath, req.FileName, "glusterd-sock", glusterdSockpath)
+		} else {
+			options = append(options, operation, brickPath, "glusterd-sock", glusterdSockpath)
+		}
+
+	}
+	_, err = runGlfshealBin(volname, options)
```

Comment:
Don't we have to error out  when splitbrain resolution is already started? 
This is missed in the testcase as well. Can you please add that as well?

Reply:
Didn't get what you are trying to state here ?

Reply:
Do we support issuing splitbrain bigger-file/lastest-mtime or any splitbrain command on a second file when the first file is already being resolved? 
If we don't support it we need to check if its already running and error out asking the user to wait.

Reply:
@harigowtham Can't really do anything in here. The error has be raised in the glfsheal binary. Not entirely sure if this is the case in glfsheal right now.

Reply:
okay